platform-service/src/unstract/platform_service/helper/cost_calculation.py

```diff
@@ -49,24 +76,47 @@ def calculate_cost(
         try:
-            # File does not exist, fetch JSON data from API
-            if not os.path.exists(self.file_path):
-                return self._fetch_and_save_json()
+            if check_feature_flag_status(FeatureFlag.REMOTE_FILE_STORAGE):
+                # File does not exist, fetch JSON data from API
+                if not self.file_storage.exists(self.file_path):
+                    return self._fetch_and_save_json()
 
-            file_mtime = os.path.getmtime(self.file_path)
-            file_expiry_date = datetime.fromtimestamp(file_mtime) + timedelta(
-                days=self.ttl_days
-            )
-            if datetime.now() < file_expiry_date:
-                app.logger.info(f"Reading model token data from {self.file_path}")
-                # File exists and TTL has not expired, read and return content
-                with open(self.file_path, encoding="utf-8") as f:
-                    return json.load(f)
+                file_mtime = self.file_storage.modification_time(self.file_path)
+                file_expiry_date = file_mtime + timedelta(days=self.ttl_days)
+                file_expiry_date_utc = file_expiry_date.replace(tzinfo=timezone.utc)
+                now_utc = datetime.now().replace(tzinfo=timezone.utc)
+
+                if now_utc < file_expiry_date_utc:
+                    app.logger.info(f"Reading model token data from {self.file_path}")
+                    # File exists and TTL has not expired, read and return content
+                    file_contents = self.file_storage.read(
+                        self.file_path, mode="r", encoding="utf-8"
+                    )
+                    return json.loads(file_contents)
+                else:
+                    # TTL expired, fetch updated JSON data from API
+                    return self._fetch_and_save_json()
             else:
-                # TTL expired, fetch updated JSON data from API
-                return self._fetch_and_save_json()
+                # File does not exist, fetch JSON data from API
+                if not os.path.exists(self.file_path):
+                    return self._fetch_and_save_json()
+
+                file_mtime = os.path.getmtime(self.file_path)
+                file_expiry_date = datetime.fromtimestamp(file_mtime) + timedelta(
+                    days=self.ttl_days
+                )
+                if datetime.now() < file_expiry_date:
+                    app.logger.info(f"Reading model token data from {self.file_path}")
+                    # File exists and TTL has not expired, read and return content
+                    with open(self.file_path, encoding="utf-8") as f:
+                        return json.load(f)
+                else:
+                    # TTL expired, fetch updated JSON data from API
+                    return self._fetch_and_save_json()
         except Exception as e:
-            app.logger.error("Error in calculate_cost: %s", e)
+            app.logger.error(
```

Comment:
@gaya3-zipstack at places where we catch, log and suppress the exception can we either
1. Log it as a warning and not an error OR
1. Explicitly specify in the log message that we'll be continuing the processing / execution

This will help us while reading the service's logs and won't throw us off while encountering such errors

Reply:
Done. CHanged pre-existing code.